Client/SMTP_client.cpp

```diff
@@ -0,0 +1,1006 @@
+		encoded_login.append("\r\n");
+		SendData(encoded_login);		
+
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_AUTH_LOGIN)
+		{
+			LOG_ERROR << "Server responded with error to user login.";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_AUTH_LOGIN_ANSWER);
+		}
+		FlushBuffer();
+
+		if (m_password.empty())
+		{
+			LOG_ERROR << "User password was not specified.";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::UNDEF_USER_PASSWORD);
+		}
+		
+		std::string encoded_password = base64_encode((const unsigned char*)m_password.c_str(), (u_int)m_password.size());
+		encoded_password.append("\r\n");
+		SendData(encoded_password);
+		
+
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_AUTH_SUCCESSFUL)
+		{
+			LOG_ERROR << "Server responded with error to user authorization.";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_AUTHORIZATION_FAILED);
+		}
+		FlushBuffer();
+
+		SendMailFrom();
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_OKAY)
+		{
+			LOG_ERROR << "Server responded with error to MAIL FROM:";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_MAIL_FROM_ANSWER);
+		}
+		FlushBuffer();
+
+		SendRcptTo();	
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_OKAY)
+		{
+			LOG_ERROR << "Server responded with error to RCPT TO:";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_RCPT_TO_ANSWER);
+		}
+		FlushBuffer();
+
+		const std::string DATA = "DATA\r\n";
+		SendData(DATA);
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_START_MAIL)
+		{
+			LOG_ERROR << "Server responded with error to DATA.";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_DATA_ANSWER);
+		}
+		FlushBuffer();
+
+		SendSubject();
+		SendData(m_letter_message.append("\r\n"));
+		SendEndingDot();
+
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_OKAY)
+		{
+			LOG_ERROR << "Server responded with error to message ending procedure.";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_LETTER_ANSWER);
+		}
+
+		SendQuit();
+		ReceiveData();
+		if (GetResponseCode() != (int)SMTPServerResponce::SERVER_QUIT)
+		{
+			LOG_ERROR << "Server responded with error to QUIT";
+			throw SMTPErrorClass(SMTPErrorClass::SMTPerrorEnum::SERVER_QUIT_ANSWER);
+		}
+	}
+	catch (const SMTPErrorClass&)
+	{
+		DisconnectFromServer();
+		throw;
+	}
+
+	return true;
+}
+
+int		SMTPClientClass::GetResponseCode() const
+{
+	if (m_receive_buffer.get() == nullptr)
+		return 0;
+
+	// Transform value from ASCII into integer
+	const int	TO_HUNDREDS	= 100;
+	const int	TO_TENS		= 10;
+	const char	CHAR_TO_INT = '0';
+
+	return (m_receive_buffer.get()[0] - CHAR_TO_INT) * TO_HUNDREDS + (m_receive_buffer.get()[1] - CHAR_TO_INT) * TO_TENS + m_receive_buffer.get()[2] - CHAR_TO_INT;
```

Comment:
m_receive_buffer.get()[0] . maybe instead of 0,1,2 -> #define BUFFER_HUNDREDS_PART 0.... and so on